hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/DiscoveryJoiner.java

```diff
@@ -49,7 +49,7 @@ public class DiscoveryJoiner
 
     public DiscoveryJoiner(Node node, DiscoveryService discoveryService, boolean usePublicAddress) {
         super(node);
-        this.maximumWaitingTimeBeforeJoinSeconds = node.getProperties().getInteger(WAIT_SECONDS_BEFORE_JOIN);
+        this.maximumWaitingTimeBeforeJoinSeconds = node.getProperties().getInteger(MAX_WAIT_SECONDS_BEFORE_JOIN);
```

Comment:
hello @jglick 👋 

My memory is hazy about this particular change, but I reckon that if you change this field to use `MAX_WAIT_SECONDS_BEFORE_JOIN` then it will increase the initial startup time from +-5s to +-20s for any discovery strategy except UDP and static TCPs. That's probably not what you want. I agree the field name is misleading, it should have been called just "waitingTimeBeforeJoinSeconds` or so.

Reply:
I tend to agree that the field has been mis-named here and would keep the `WAIT_SECONDS_BEFORE_JOIN` property. @hasancelik wdyt?